### PR TITLE
Add initial structure and Nuget configuration of the `Project Template` of the TodoTemplate (#1793)

### LINKED ISSUE #1793: Implement the Initial structure and Nuget configuration of the `Project Template` of the TodoTemplate
The project template of the TodoTemplate needs to be designed and developed in order to be used in the dotnet cli or visual studio.

### PR DESCRIPTION
**Pull request by MeysamBahadori**
this closes #1793

### CODE REVIEW

**Review by msynk — COMMENTED** [review]
@MeysamBahadori 
please update this PR based on our conversation :)

**Comment by MeysamBahadori** [comment]
> @MeysamBahadori please update this PR based on our conversation :)

Ok

**Comment by MeysamBahadori** [comment]
Ready for review...

**Comment by MeysamBahadori** [comment]
BlazorMode and WebAppDeploymentType symbols removed.

**Comment by msynk** [comment]
@MeysamBahadori 
plz resolve the conflicts

**Comment by MeysamBahadori** [comment]
Plz review again...

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Api/Startup/Middlewares.cs [hunk 1]
```diff
@@ -45,6 +45,7 @@ public static void Use(IApplicationBuilder app, IHostEnvironment env)
         {
             endpoints.MapDefaultControllerRoute();
 
+
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Api/Startup/Middlewares.cs:48]: plz remove this change

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Shared/Infra/BuildConfigurationModeDetector.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+
+        public virtual BuildConfigurationMode Mode
+        {
+            get
+            {
+#if DEBUG
+                return BuildConfigurationMode.Debug;
+                #else
```
Comment (msynk) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Shared/Infra/BuildConfigurationModeDetector.cs:32]: plz correct the code formatting in this line

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Extensions/IServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -20,7 +21,6 @@ public static IServiceCollection AddTodoTemplateAppServices(this IServiceCollect
             return httpClient;
         });
 #endif
-
```
Comment (msynk) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Extensions/IServiceCollectionExtensions.cs:23]: plz remove this change

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Program.cs [hunk 4]
```diff
@@ -50,4 +50,5 @@ public static WebApplication CreateHostBuilder(string[] args)
         return app;
     }
 #endif
+
```
Comment (msynk) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Program.cs:53]: plz remove this change

File: src/Bit.sln [hunk 5]
```diff
@@ -83,6 +88,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bit.Platform.WebSite.Shared
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bit.Platform.WebSite.Web", "WebSite\Bit.Platform.WebSite\Web\Bit.Platform.WebSite.Web.csproj", "{85F4E9DD-46F4-48B2-8E34-15229700B488}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TodoTemplate.nuget", "Tooling\Bit.Tooling.Templates\TodoTemplate\TodoTemplate.nuget.csproj", "{95571715-5D44-478C-9681-A35543A2A5F5}"
```
Comment (msynk) [inline src/Bit.sln:91]: isn't this project renamed to `TodoTemplate.ProjectTemplate.csproj`?

**Review by msynk — CHANGES_REQUESTED** [review]

**Review by msynk — APPROVED** [review]

**Pull request merged**